builtin/game/chatcommands.lua

```diff
@@ -15,6 +15,29 @@ function core.register_chatcommand(cmd, def)
 	core.registered_chatcommands[cmd] = def
 end
 
+function core.unregister_chatcommand(name)
+	if not core.registered_chatcommands[name] then
+		core.log("warning", "Not unregistering chatcommand " ..name..
```

Comment:
It works just fine for me tested in the experimental mod in miminal with the exact example code provided in OP.